### ISSUE TITLE
Modify Tags on azurerm_sql_database

### ISSUE THREAD

**Issue by makaanneo** [hashicorp/terraform-provider-azurerm#3991]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform v0.12.5
"azurerm" (terraform-providers/azurerm) 1.31.0

### Affected Resource(s)

* `azurerm_sql_database`

### Terraform Configuration Files

```hcl
locals {
  deployment_time = timestamp()
}

resource "azurerm_sql_database" "tenant" {
  name                = "db"
  resource_group_name = "db-rg"
  location            = "germanycentral"
  server_name         = "sqlsvr"
  elastic_pool_name   = "ep"
  create_mode         = "Copy"
  source_database_id  = "sourceId to valid database"

  tags = {
    environment          = "test1"
    misc                 = "test2"
    tenant_guid          = "test3"
    tenant_surrogate     = "test3"
    test = local.deployment_time
  }

  lifecycle {
    ignore_changes  = [ source_database_id, create_mode, tags["test"]]
  }
}
```

### Panic Output

Api output 
Error: Error issuing create/update request for SQL Database "db" (Resource Group "db-rg", Server "sqlsvr"): sql.DatabasesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="Failed" Message="The async operation failed." InnerError={"unmarshalError":"json: cannot unmarshal array into Go struct field serviceError2.details of type map[string]interface {}"} AdditionalInfo=[{"code":"BadRequest","details":[],"innererror":[],"message":"The destination database name 'db' already exists on the server 'sqlsvr'.","target":null}]

### Expected Behavior

ignore_changes on certain tags works
Update tags on database with copy works

### Actual Behavior

ignore_changes on certain tags does not work.
Update a tag on a sql database in azure does not work

### Steps to Reproduce

1. `terraform apply` -> create resource
2.  `terraform apply` -> modify resource

### Important Factoids

Running in a Azure Germany Environment

### References

* #0000

**Comment by akonrath** [comment]
Hi @makaanneo ! Are you still experiencing this issue on the latest terraform and azurerm provider versions? I attempted to replicate this issue, and was unable to do so on the latest versions.

I tested adding and removing tags from the azurerm_sql_database resource, and encountered no issues; tags were successfully updated each time.

Local configuration:
```
$ terraform --version
Terraform v0.12.24
+ provider.azurerm v2.6.0
```

Terraform code:
```
locals {
  suffix          = "akonrath"
  deployment_time = timestamp()
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "acceptanceTestResourceGroup1"
  location = "West US"
}

resource "azurerm_sql_server" "example" {
  name                         = "mysqlserver${local.suffix}"
  resource_group_name          = azurerm_resource_group.example.name
  location                     = "West US"
  version                      = "12.0"
  administrator_login          = "4dm1n157r470r"
  administrator_login_password = "4-v3ry-53cr37-p455w0rd"

  tags = {
    environment = "production"
  }
}

resource "azurerm_storage_account" "example" {
  name                     = "examples${local.suffix}"
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_sql_database" "example" {
  name                = "mysqldatabase"
  resource_group_name = azurerm_resource_group.example.name
  location            = "West US"
  server_name         = azurerm_sql_server.example.name

  extended_auditing_policy {
    storage_endpoint                        = azurerm_storage_account.example.primary_blob_endpoint
    storage_account_access_key              = azurerm_storage_account.example.primary_access_key
    storage_account_access_key_is_secondary = true
    retention_in_days                       = 6
  }

  tags = {
    environment      = "test1"
    misc             = "test2"
    tenant_guid      = "test3"
    tenant_surrogate = "test3"
    test             = local.deployment_time
  }

  lifecycle {
    ignore_changes = [source_database_id, create_mode, tags["test"]]
  }
}
```

**Comment by makaanneo** [comment]
Hi, 
sorry for the delay ..., I tried it with a modified version of your and mine example.

```
terraform -version
Terraform v0.12.24
+ provider.azurerm v2.7.0
```

reproduced on Azure Public Region "Germany West Central"

```
locals {
  suffix          = "makaanneo"
  deployment_time = timestamp()
}

provider "azurerm" {
  features {}
}

resource "random_string" "random" {
  length = 15
  special = false
  lower = true
  upper = false
}


resource "azurerm_resource_group" "example" {
  name     = "${random_string.random.result}rg${local.suffix}"
  location = "Germany West Central"
}

resource "azurerm_sql_server" "example" {
  name                         = "${random_string.random.result}srv"
  resource_group_name          = azurerm_resource_group.example.name
  location                     = azurerm_resource_group.example.location
  version                      = "12.0"
  administrator_login          = "4dm1n157r470r"
  administrator_login_password = "4-v3ry-53cr37-p455w0rd"

  tags = {
    environment = "production"
  }
}

resource "azurerm_sql_elasticpool" "example" {
  name                = "${random_string.random.result}ep"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  server_name         = azurerm_sql_server.example.name
  edition             = "Basic"
  dtu                 = 50
  pool_size           = 5000
}

resource "azurerm_storage_account" "example" {
  name                     = "${random_string.random.result}db"
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_sql_database" "copy_source" {
  name                = "${random_string.random.result}dbcopyfrom"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  server_name         = azurerm_sql_server.example.name
  elastic_pool_name   = azurerm_sql_elasticpool.example.name

  extended_auditing_policy {
    storage_endpoint                        = azurerm_storage_account.example.primary_blob_endpoint
    storage_account_access_key              = azurerm_storage_account.example.primary_access_key
    storage_account_access_key_is_secondary = true
    retention_in_days                       = 6
  }
}

resource "azurerm_sql_database" "example" {
  name                = "${random_string.random.result}target"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  server_name         = azurerm_sql_server.example.name
  elastic_pool_name   = azurerm_sql_elasticpool.example.name
  create_mode         = "Copy"
  source_database_id  = azurerm_sql_database.copy_source.id

  extended_auditing_policy {
    storage_endpoint                        = azurerm_storage_account.example.primary_blob_endpoint
    storage_account_access_key              = azurerm_storage_account.example.primary_access_key
    storage_account_access_key_is_secondary = true
    retention_in_days                       = 6
  }

  tags = {
    environment      = "test1"
    misc             = "test2"
    tenant_guid      = "test3"
    tenant_surrogate = "test3"
    not_ignored = local.deployment_time
    test             = local.deployment_time
  }

  lifecycle {
    ignore_changes = [source_database_id, create_mode, tags["test"]]
  }
}

```

1. terraform init
2. terraform apply
3. terraform apply (without any change, only tag value not_ignored will be updated)

## Result
```
terraform apply --auto-approve
random_string.random: Refreshing state... [id=ay33pl55vxo37mn]
azurerm_resource_group.example: Refreshing state... [id=/subscriptions/subscription/resourceGroups/ay33pl55vxo37mnrgmakaanneo]
azurerm_sql_server.example: Refreshing state... [id=/subscriptions/subscription/resourceGroups/ay33pl55vxo37mnrgmakaanneo/providers/Microsoft.Sql/servers/ay33pl55vxo37mnsrv]
azurerm_storage_account.example: Refreshing state... [id=/subscriptions/subscription/resourceGroups/ay33pl55vxo37mnrgmakaanneo/providers/Microsoft.Storage/storageAccounts/ay33pl55vxo37mndb]
azurerm_sql_elasticpool.example: Refreshing state... [id=/subscriptions/subscription/resourceGroups/ay33pl55vxo37mnrgmakaanneo/providers/Microsoft.Sql/servers/ay33pl55vxo37mnsrv/elasticPools/ay33pl55vxo37mnep]
azurerm_sql_database.copy_source: Refreshing state... [id=/subscriptions/subscription/resourceGroups/ay33pl55vxo37mnrgmakaanneo/providers/Microsoft.Sql/servers/ay33pl55vxo37mnsrv/databases/ay33pl55vxo37mndbcopyfrom]
azurerm_sql_database.example: Refreshing state... [id=/subscriptions/subscription/resourceGroups/ay33pl55vxo37mnrgmakaanneo/providers/Microsoft.Sql/servers/ay33pl55vxo37mnsrv/databases/ay33pl55vxo37mntarget]
azurerm_sql_database.example: Modifying... [id=/subscriptions/subscription/resourceGroups/ay33pl55vxo37mnrgmakaanneo/providers/Microsoft.Sql/servers/ay33pl55vxo37mnsrv/databases/ay33pl55vxo37mntarget]

Error: Error issuing create/update request for SQL Database "ay33pl55vxo37mntarget" (Resource Group "ay33pl55vxo37mnrgmakaanneo", Server "ay33pl55vxo37mnsrv"): sql.DatabasesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="Failed" Message="The async operation failed." InnerError={"unmarshalError":"json: cannot unmarshal array into Go struct field serviceError2.details of type map[string]interface {}"} AdditionalInfo=[{"code":"BadRequest","details":[],"innererror":[],"message":"The destination database name 'ay33pl55vxo37mntarget' already exists on the server 'ay33pl55vxo37mnsrv'.","target":null}]

  on main.tf line 69, in resource "azurerm_sql_database" "example":
  69: resource "azurerm_sql_database" "example" {


```

**Comment by neil-yechenwei** [comment]
@makaanneo, Thanks for opening this issue. I assume it's expected behavior. Because `not_ignored = local.deployment_time` is using `timestamp()` and this function would get current time so that the value is dynamic and different on each time.

**Comment by tombuildsstuff** [comment]
Looks like the Create and Update may need separating here, so that we're only Restoring during a Create, which is I believe the root-cause here

**Comment by waynejnicklin** [comment]
I have a similar issue... I added some tags at the database level, using the terraform below. The plan file correctly showed the intention to add the tags, but the apply had no effect. However, the state file was updated to include the new tags, so the next time the pipeline ran the changes did not appear in the plan file.

    tags = {
        Environment       = var.environment_name
        Product           = var.product_name
        DefaultScale      = var.sql_scale
        ScaledownSchedule = var.sql_scaledown_schedule
    }

I'm not using ignore_changes, just simply trying to add new tags.

Thanks,
Wayne
